### PR TITLE
:package: Ignore `.phpstorm.meta.php` because this is not PHP file

### PR DESCRIPTION
**Pull request by andrew-demb**
Replaces https://github.com/rectorphp/rector-src/pull/6593

Refs: https://github.com/rectorphp/rector-src/pull/6593#issuecomment-2543992451

### CODE REVIEW

File: src/Configuration/RectorConfigBuilder.php [hunk 1]
```diff
@@ -415,7 +415,11 @@ function (string $string): string {
             ->depth(0)
             ->ignoreDotFiles(false)
             ->name('*.php')
-            ->name('.*.php');
+            ->name('.*.php')
+            // this file cannot be interpreted as PHP file
+            // https://www.jetbrains.com/help/phpstorm/ide-advanced-metadata.html#expected-arguments
+            ->notName('.phpstorm.meta.php')
+        ;
```
Comment (samsonasik) [inline src/Configuration/RectorConfigBuilder.php:422]: I think we should not need to enable dot files by default, eg:

```php
public function withRootFiles(bool $includeDotFiles = false)
```

ensure be safe by default, while still hardcoded this obvious `.phpstorm.meta.php` so no change by default.

Then you can add condition:

```php
if ($includeDotFiles) {
     $rootPhpFilesFinder->ignoreDotFiles(false);
}

Comment (andrew-demb) [reply src/Configuration/RectorConfigBuilder.php:422]: I'm OK with all options (enable for all dotfiles except known risky, enable for known dotfiles, provide an argument, etc)

Comment (samsonasik) [reply src/Configuration/RectorConfigBuilder.php:422]: I am for:

- disable by default (make optional)
- still exclude `.phpstorm.meta.php` as known risky

**Comment by samsonasik** [comment]
Let's merge to keep forward and ensure other PR on sync not hung up on run rector.

Let's see how next improvement will happen

**Comment by samsonasik** [comment]
Thank you @andrew-demb

File: src/Configuration/RectorConfigBuilder.php (hunk 1 above)
Comment (andrew-demb) [reply src/Configuration/RectorConfigBuilder.php:422]: @TomasVotruba, what do you think about this?

**Pull request merged**